File: repository-example/src/test/java/nl/haploid/octowight/TestData.java
package nl.haploid.octowight;

import nl.haploid.octowight.repository.BookDmo;

import java.util.Random;
import java.util.UUID;

public class TestData {

	public static BookDmo bookDmo(final String genre) {
		return bookDmo(genre, nextLong());
	}

	public static BookDmo bookDmo(final String genre, final long id) {
		final BookDmo book = new BookDmo();
		book.setId(id);
		book.setGenre(genre);
		book.setTitle(title());
		return book;
	}

	public static long nextLong() {
		return new Random().nextLong();
	}

	public static String title() {
		return UUID.randomUUID().toString();
	}
}

File: sample-resource-builder/src/test/scala/nl/haploid/octowight/sample/TestData.scala
package nl.haploid.octowight.sample

import java.util.{Random, UUID}
import nl.haploid.octowight.registry.data.{ResourceMessage, Atom, ResourceIdentifier, ResourceRoot}
import nl.haploid.octowight.sample.repository.RoleDmo

object TestData {

  def atom = new Atom(id = nextLong, origin = nextString, category = nextString)

  def nextLong = new Random().nextLong

  def nextString = UUID.randomUUID.toString

  def resource = new ResourceMessage(resourceIdentifier = resourceIdentifier, model = nextString, tombstone = false)

  def resourceIdentifier = new ResourceIdentifier(collection = nextString, id = nextLong)

  def resourceRoot =
    new ResourceRoot(resourceId = nextLong, resourceCollection = "captain", root = atom, version = nextLong)

  def roleDmo = new RoleDmo
}
